projects/client/RabbitMQ.Client/src/client/impl/CommandAssembler.cs

```diff
@@ -98,20 +98,19 @@ public Command HandleFrame(Frame f)
                 }
                 case AssemblyState.ExpectingContentBody:
                 {
-                    if (f.Type != Constants.FrameBody)
+                    if (!f.IsBody())
                     {
                         throw new UnexpectedFrameException(f);
                     }
-                    byte[] fragment = f.Payload;
```

Comment:
Is there real value in removing this variable? I would have thought (hoped) the compiler would optimise this away anyway.